types/react-stripe-elements/index.d.ts

```diff
@@ -15,102 +15,99 @@
-	type HTMLStripeElement = stripe.elements.Element;
-
-	/**
-	 * There's a bug in @types/stripe which defines the property as
-	 * `declined_code` (with a 'd') but it's in fact `decline_code`
-	 */
-	type PatchedTokenResponse = TokenResponse & {
-		error?: { decline_code?: string };
-	};
-
-	interface StripeProviderOptions {
-		stripeAccount?: string;
-	}
-	type StripeProviderProps = { apiKey: string; stripe?: never; } & StripeProviderOptions | { apiKey?: never; stripe: stripe.Stripe | null; } & StripeProviderOptions;
-
-	interface StripeProps {
-		createSource(sourceData?: SourceOptions): Promise<SourceResponse>;
-		createToken(options?: TokenOptions): Promise<PatchedTokenResponse>;
-		paymentRequest: stripe.Stripe['paymentRequest'];
-		createPaymentMethod(
-			paymentMethodType: stripe.paymentMethod.paymentMethodType,
-			data?: stripe.CreatePaymentMethodOptions,
-		): Promise<stripe.PaymentMethodResponse>;
-		handleCardPayment(
-			clientSecret: string,
-			options?: stripe.HandleCardPaymentOptions
-		): Promise<stripe.PaymentIntentResponse>;
-		handleCardSetup(
-			clientSecret: string,
-			data?: stripe.HandleCardSetupOptions
-		): Promise<stripe.PaymentIntentResponse>;
-	}
-
-	interface InjectOptions {
-		withRef?: boolean;
-	}
-
-	interface InjectedStripeProps {
-		stripe?: StripeProps;
-	}
-
-	interface ElementProps extends ElementsOptions {
-		id?: string;
-
-		className?: string;
-
-		elementRef?(ref: any): void;
-
-		onChange?(event: ElementChangeResponse): void;
-
-		onBlur?(event: ElementChangeResponse): void;
-
-		onFocus?(event: ElementChangeResponse): void;
-
-		onReady?(el: HTMLStripeElement): void;
-	}
+    type ElementChangeResponse = stripe.elements.ElementChangeResponse;
+    type ElementsOptions = stripe.elements.ElementsOptions;
+    type TokenOptions = stripe.TokenOptions;
+    type TokenResponse = stripe.TokenResponse;
+    type SourceResponse = stripe.SourceResponse;
+    type SourceOptions = stripe.SourceOptions;
+    type HTMLStripeElement = stripe.elements.Element;
+
+    /**
+     * There's a bug in @types/stripe which defines the property as
+     * `declined_code` (with a 'd') but it's in fact `decline_code`
+     */
+    type PatchedTokenResponse = TokenResponse & {
+        error?: { decline_code?: string };
+    };
+
+    interface StripeProviderOptions {
+        stripeAccount?: string;
+    }
+    type StripeProviderProps =
+        | { apiKey: string; stripe?: never } & StripeProviderOptions
+        | { apiKey?: never; stripe: stripe.Stripe | null } & StripeProviderOptions;
+
+    // From https://stripe.com/docs/stripe-js/reference#element-types
+    type TokenType = 'card' | 'cardNumber' | 'cardExpiry' | 'cardCvc' | 'paymentRequestButton' | 'iban' | 'idealBank';
+
+    interface StripeProps {
+        createSource(sourceData?: SourceOptions): Promise<SourceResponse>;
+        createToken(options?: TokenOptions & { type?: TokenType }): Promise<PatchedTokenResponse>;
```

Comment:
stylistically, I'd prefer to see react-stripe-elements' TokenOptions type change, like so:

```ts
type TokenOptions = stripe.TokenOptions & { type?: TokenType };
```

And then this line wouldn't have to change.